Source/WebGPU/WebGPU/Texture.mm

```diff
@@ -1339,6 +1339,46 @@ static bool textureViewFormatCompatible(WGPUTextureFormat format1, WGPUTextureFo
     return true;
 }
 
+bool Device::validateCreateIOSurfaceBackedTexture(const WGPUTextureDescriptor& descriptor, const Vector<WGPUTextureFormat>& viewFormats, IOSurfaceRef backing)
```

Comment:
Are we confident that the implementation of this function is comprehensive?

Reply:
I'm _fairly_ confident since I strictly followed what Metal people told me about.

Reply:
Now to think about it, I'd like to test more thing in the `IOSurfaceRef`

Reply:
I added two more checks: pixel format and non-planar.